buildlib/az-helpers.sh

```diff
@@ -247,3 +257,149 @@ get_arch() {
         echo "$arch"
     fi
 }
+
+declare -A COLORS=(
+    [RED]='\033[0;31m'
+    [GREEN]='\033[0;32m'
+    [BLUE]='\033[0;34m'
+    [CYAN]='\033[0;36m'
+    [NC]='\033[0m' # No Color
+)
+
+function log_script_info() {
+    echo -e "${COLORS[CYAN]}Script Information${COLORS[NC]}"
+    echo -e "${COLORS[BLUE]}----------------------------${COLORS[NC]}"
+    echo -e "${COLORS[GREEN]}Script:${COLORS[NC]} $0"
+    echo -e "${COLORS[GREEN]}Working Directory:${COLORS[NC]} $(pwd)"
+    echo -e "${COLORS[GREEN]}User:${COLORS[NC]} $(whoami)"
+}
+
+function log_machine_info() {
+    echo -e "\n${COLORS[CYAN]}Machine Information${COLORS[NC]}"
+    echo -e "${COLORS[BLUE]}----------------------------${COLORS[NC]}"
+    if [ -f /.dockerenv ]; then
+        echo -e "${COLORS[GREEN]}Running in container:${COLORS[NC]} Yes"
+        echo -e "${COLORS[GREEN]}Agent Machine Name:${COLORS[NC]} $AGENT_MACHINENAME"
+        echo -e "${COLORS[GREEN]}Container Image:${COLORS[NC]} $(awk -F= '/^PRETTY_NAME=/ {gsub(/"/, "", $2); print $2}' /etc/os-release)"
+        log_docker_variables
+    else
+        echo -e "${COLORS[GREEN]}Running in container:${COLORS[NC]} No"
+        echo -e "${COLORS[GREEN]}Host Machine:${COLORS[NC]} $(uname -n)"
+    fi
+    echo -e "${COLORS[GREEN]}OS:${COLORS[NC]} $(uname -s) ($(uname -o))"
+    echo -e "${COLORS[GREEN]}Kernel:${COLORS[NC]} $(uname -r) ($(uname -v))"
+    echo -e "${COLORS[GREEN]}Architecture:${COLORS[NC]} $(uname -m)"
+}
+
+function log_docker_variables() {
+    if env | grep -q 'DOCKER_OPT_'; then
+        echo -e "${COLORS[GREEN]}Docker Variables:${COLORS[NC]}"
+        for var in DOCKER_OPT_VOLUMES DOCKER_OPT_IB DOCKER_OPT_GPU DOCKER_OPT_ARGS; do
```

Comment:
Is that all possible docker related env variables? Maybe just echo content from all the values found by grep as you do for UCX variables?

Reply:
Yes, you can see it in `main.yml`